service/controller/clusterapi/v29/resource/tcnp/create.go

```diff
@@ -209,7 +209,7 @@ func (r *Resource) newIAMPolicies(ctx context.Context, cl v1alpha1.Cluster, md v
 			ID: key.MachineDeploymentID(&md),
 		},
 		RegionARN: key.RegionARN(cl),
-		S3Bucket:  key.BucketName(cl, cc.Status.TenantCluster.AWSAccountID),
+		S3Bucket:  key.BucketName(&cl, cc.Status.TenantCluster.AWSAccountID),
```

Comment:
Let's use the machine deployment here instead of the cluster so we can narrow down its usage as discussed.